script/deploy/DeployTestnet.s.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import {console} from "@forge-std/console.sol";
+import {Script} from "@forge-std/Script.sol";
+import {Addresses} from "@proposals/Addresses.sol";
+import {Proposal} from "@proposals/proposalTypes/Proposal.sol";
+
+import {zip000} from "@proposals/zips/zip000.sol";
+import {zip001} from "@proposals/zips/zip001.sol";
+import {zip002} from "@proposals/zips/zip002.sol";
+import {zip003} from "@proposals/zips/zip003.sol";
+import {zip004} from "@proposals/zips/zip004.sol";
+import {zip005} from "@proposals/zips/zip005.sol";
+import {zip006} from "@proposals/zips/zip006.sol";
+import {zip007} from "@proposals/zips/zip007.sol";
+import {zip008} from "@proposals/zips/zip008.sol";
+
+/*
+How to use:
+forge script script/deploy/DeployTestnet.s.sol:DeployTestnet \
+    -vvvv \
+    --rpc-url $ETH_RPC_URL \
+    --broadcast \
+    --private-key <KEY>
+Remove --broadcast and --private-key if you want to try locally first, without paying any gas.
+*/
+
+contract DeployTestnet is Script {
+    uint256 public privateKey;
+
+    Addresses addresses;
+    Proposal[] public proposals;
+
+    function setUp() public {
+        privateKey = vm.envUint("DEPLOYER_PRIVATE_KEY");
+
+        string memory environment = vm.envOr("ENVIRONMENT", string("localnet"));
+        string memory addressPath = string(abi.encodePacked("proposals/Addresses/", environment, ".json"));
+        addresses = new Addresses(addressPath);
+        addresses.resetRecordingAddresses();
+
+        // Load proposals
+        proposals.push(Proposal(address(new zip000()))); /// Genesis token proposal
+        proposals.push(Proposal(address(new zip001()))); /// Wearables, Core, ADMIN_MULTISIG proposal
+        proposals.push(Proposal(address(new zip002()))); /// Timelock proposal
+        proposals.push(Proposal(address(new zip003()))); /// CGv1 proposal
+        proposals.push(Proposal(address(new zip004()))); /// TokenIds, MaxSupply and Capsule settings proposal
+        proposals.push(Proposal(address(new zip005()))); /// MaxSupply settings proposal
+        proposals.push(Proposal(address(new zip006()))); /// MaxSupply settings proposal
+        proposals.push(Proposal(address(new zip007()))); /// MaxSupply settings proposal
+        proposals.push(Proposal(address(new zip008()))); /// MaxSupply settings proposal
+    }
+
+    function run() public {
+        for (uint256 i = 0; i < proposals.length; i++) {
+            string memory name = proposals[i].name();
+            console.log("Proposal", name, "deploy()");
+            addresses.resetRecordingAddresses();
+
+            address deployer = vm.addr(privateKey);
+            // Run the deploy for testing only workflow
+            proposals[i].deployForTestingOnly(addresses, deployer);
```

Comment:
The system was already built like this. This was done because the admin role is revoked from deployer on mainnet in zip001 [here](https://github.com/ZTX-Foundation/tuxedo/blob/6530048b0400d80563489671319e184610fdbd3e/proposals/zips/zip001.sol#L67) while that is not done on testnets.

I am working on making it a single contract in a separate PR.

Reply:
Can't we just update this PR instead?